### ISSUE TITLE
icons-angular accessibility violation: svg with img role must have a nonempty alternate text or a label

### ISSUE THREAD

**Issue by jdcai** [carbon-design-system/carbon-elements#502]
## Detailed description

I'm using the dynamic assessment plugin to check for accessibility violations and I'm getting  

The WAI-ARIA widget with img role must have a nonempty alternate text or a label.

The violation is on the svg element with the role="img". The violation says it should have an alt attribute on it. I tried adding an aria label but I still get the violation. If I add an alt attribute though the developer tools and rerun DAP then I don't get the violation.

> Is this a feature request (new component, new icon), a bug, or a general issue?

bug

> Is this issue related to a specific component?

icons-angular

> What did you expect to happen? What happened instead? What would you like to see changed?

There should be a way to add alt attribute to the svg similar to how the other attributes like class or aria label get added.

> What browser are you working in?

chrome

> What version of the Carbon Design System are you using?

9.40.0

**Comment by jdcai** [comment]
If I have title= in the tag like ` <ibm-icon-information-filled20 title="'test'">      </ibm-icon-information-filled20>` then a role="img" is added to the svg and causes the accessibility violation.
Without the title= it doesn't get a role="img" added and doesn't have the accessibility violation.

**Comment by joshblack** [comment]
cc @cal-smith

**Comment by cal-smith** [comment]
Weird, I'll have to dig into this 🤔 I know most of our a11y property setting is handled by `getAttributes` from `icon-helpers`

**Comment by joshblack** [comment]
@jdcai things get a little tricky here, unfortunately, and it's definitely on us to add better documentation here.

When setting `title` on the icon, we enter into this branch in the code: https://github.com/carbon-design-system/carbon-elements/blob/master/packages/icon-helpers/src/getAttributes.js#L41

What this means is that we need the role of the SVG to be img and then the consumer will need to follow one of several patterns to make it fully accessible: https://www.deque.com/blog/creating-accessible-svgs/

The most reliable patterns seem to be pattern 5 for simple labels and pattern 11 for complex labels where descriptions also need to be offered.

Unfortunately we can't do a pattern by default as different teams may require different features depending on what they target, although if we can come up with a decent baseline then we should totally do it 😄

**Comment by joshblack** [comment]
Another thing to add to this is that `alt` is not a valid attribute on the `<svg>` tag so DAP might need to be updated, here.

**Comment by dakahn** [comment]
Closing as resolved 🏄🏽‍♂️
